02_bubble/pokkalar/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubbleSort(s []int) []int {
+	c := make([]int, len(s))
+	copy(c, s)
+	for i := 0; i < len(c); i++ {
+		swapped := false
+		for j := 0; j < len(c)-1; j++ {
+			if c[j] > c[j+1] {
+				c[j+1], c[j] = c[j], c[j+1]
+				swapped = true
+			}
+		}
+		if !swapped {
+			break
+		}
+	}
+	return c
+}
+
+func insertionSort(s []int) []int {
+	i := make([]int, len(s))
+	copy(i, s)
+	for j := 1; j < len(i); j++ {
+		for k := j; k > 0; k-- {
+			if s[k] < s[k-1] {
```

Comment:
`s` is the input, `i` is the copy, I think you meant to perform the sort on `i`, would be worth adding a check to your tests that makes sure the original array has not changed.

Reply:
Thanks for reviewing the changes fixed the review comments